### DISCUSSION TITLE
Modelling of a electric UAV?

### DISCUSSION THREAD

**Discussion by prashant19012002** [JSBSim-Team/jsbsim#969] [Q&A]
Sorry i am new to JSBSim and excuse me if this problem seems dumb. i tried to build a electric UAV model(named as [ test aicraft](https://drive.google.com/file/d/1h5hb5e8_tg4n1-bZsvknjNolUI5VKBvE/view?usp=sharing)) and i tried to run it for various cases specifically a full trim case here. for some reasons i am not getting the output correctly even if i want it to run for 100s it just stops at 2 or 5 sec. (the scripts file is named as [cruise_script](https://drive.google.com/file/d/1Ua6nXV1J8Y-JygXhB7mZ-6NVZzhSKfn7/view?usp=drive_link))
[test_aircraft.zip](https://github.com/JSBSim-Team/jsbsim/files/12712390/test_aircraft.zip)

**Comment by seanmcleod** [comment]
Does the trim succeed?

**Reply by prashant19012002** [reply]
it say's full trim but it stops their. and the end time in csv file is just 2.1s
![image](https://github.com/JSBSim-Team/jsbsim/assets/145685228/a896c3da-5043-412b-a65a-05db8af43eb1)

**Comment by seanmcleod** [comment]
Hmm, something going wrong with the trim I suspect. For example the trim routine should print out the trim results, whether it's sucessful or not. For example when running the `737_cruise.xml` script which includes a trim event.

```sh
C:\source\jsbsim> .\Debug\JSBSim.exe --script=scripts\737_cruise.xml
....
....

Start: Monday September 25 2023 16:10:21 (HH:MM:SS)

Repeating Notify (Event 2) executed at time: 5.008133
    propulsion/engine[0]/n2 = 60.000000
    propulsion/engine[1]/n2 = 60.000000
    propulsion/engine[0]/thrust-lbs = 381.260934
    propulsion/engine[1]/thrust-lbs = 381.260934
    propulsion/engine[0]/thrust-N = 1695.933275
    propulsion/engine[1]/thrust-N = 1695.933275
    position/h-agl-ft = 29831.535884
    velocities/vc-kts = 282.517887
    velocities/vc-fps = 476.836514
    velocities/vt-fps = 741.851429
    attitude/phi-rad = -0.000667
    attitude/theta-rad = -0.093210
    attitude/psi-rad = 3.927292

  Full Trim

  Trim successful
  Trim Results:
       Angle of Attack:   2.28  wdot:  1.15e-04 Tolerance: 1e-03  Passed
              Throttle:   0.93  udot: -6.26e-05 Tolerance: 1e-03  Passed
            Pitch Trim:  -0.17  qdot: -1.57e-10 Tolerance: 1e-04  Passed
            Roll Angle:  -0.14  vdot: -1.55e-10 Tolerance: 1e-03  Passed
              Ailerons:  -0.00  pdot: -1.16e-08 Tolerance: 1e-04  Passed
                Rudder:  -0.00  rdot:  5.13e-19 Tolerance: 1e-04  Passed
```

Also take a look at your output, the engine doesn't appear to be running and isn't producing any thrust.

**Reply by prashant19012002** [reply]
i checked trough engine files of various aircrafts and the way the propeller files where mentioned and i have written my files according do that only. and since its loading i am not able to debug it. i tried doing it with x8 but it loaded the engine file but its trim to failed. if u can provide me reference of any other electric conventional UAV it would be helpful.

**Reply by memy009** [reply]
I worked on JSBSim also for my internship and I could have some helpful xmls for you, do you have any contact for me to send it to you??

**Reply by prashant19012002** [reply]
Hi @memy009 thanks for reply you can send it to my mail id prashantsattaraggi@gmail.com or you can upload it here if possible. Thanks for your help.

**Reply by prashant19012002** [reply]
Hi @memy009 can you please share me your files it would be very helpful for me.

**Comment by seanmcleod** [comment]
> and since its loading i am not able to debug it.

What issue are you having debugging? You should be able to set breakpoints in the JSBSim source code where the trim code is executed to see what information you see in the debugger. Also in the code to set the engine running to try and see why the engine doesn't appear to be running and producing thrust.

P.S
Any reason you don't start sentences with a capital letter? Makes it more difficult to read.

**Reply by prashant19012002** [reply]
Ok @seanmcleod i will try that once. Since i am new to it coding and debugging its taking time but yeah i will try for what you have suggested and reach out. 
And sorry for making it difficult for you to understand my sentences i will ensure next time it wont be repeated.

**Reply by prashant19012002** [reply]
![image](https://github.com/JSBSim-Team/jsbsim/assets/145685228/e121cf90-d73c-4858-8fc1-faacb0f7e5b9)
I tried running my engine and thruster files with the bench electric example mentioned [here](https://github.com/mvacanti/jsbsim/blob/pr-bldc-validation/scripts/bench_electric.xml). and i was able to do this after few changes in my propeller file.
But now for the same propeller config i am getting the error mentioned in this image.
